# HG changeset patch
# User Rik <rik@octave.org>
# Date 1443211501 25200
#      Fri Sep 25 13:05:01 2015 -0700
# Node ID 8164c580922b1dab8ecfa06a7c4ef89ce04fac8c
# Parent  9502e0142c198534d78f250e61731b30d63f7e8c
plotmatrix.m: Fix regression in %!demo introduced by in-place operators.

* plotmatrix.m: Rename nargin variable to narg variable to avoid shadowing
nargin() function.

diff --git a/scripts/plot/draw/plotmatrix.m b/scripts/plot/draw/plotmatrix.m
--- a/scripts/plot/draw/plotmatrix.m
+++ b/scripts/plot/draw/plotmatrix.m
@@ -125,37 +125,38 @@ function plotmatrixdelete (h, d, ax)
     endif
   endfor
 endfunction
 
 function [h, ax, p, pax] = __plotmatrix__ (bigax, varargin)
   have_line_spec = false;
   have_hist = false;
   parent = get (bigax, "parent");
-  for i = 1 : nargin - 1
+  narg = nargin ();
+  for i = 1 : narg - 1
     arg = varargin{i};
     if (ischar (arg) || iscellstr (arg))
       [linespec, valid] = __pltopt__ ("plotmatrix", varargin{i}, false);
       if (valid)
         have_line_spec = true;
         linespec = varargin(i);
         varargin(i) = [];
-        nargin -= 1;
+        narg -= 1;
         break;
       else
         print_usage ("plotmatrix");
       endif
     endif
   endfor
 
-  if (nargin == 2)
+  if (narg == 2)
     X = varargin{1};
     Y = X;
     have_hist = true;
-  elseif (nargin == 3)
+  elseif (narg == 3)
     X = varargin{1};
     Y = varargin{2};
   else
     print_usage ("plotmatrix");
   endif
 
   if (rows (X) != rows (Y))
     error ("plotmatrix: dimension mismatch in the arguments");
